session_0/item.rb

```diff
@@ -22,15 +22,21 @@ def initialize(name: '', price: 0, quantity: 1, category_id: 4,
 
   # Returns a boolean value whether than item is discounted i.e. the
   # discount deadline has been crossed or not.
-  def discounted?
-    raise NotImplementedError # TODO
+  def discounted?()
+    return Time.now < @discount_deadline
   end
 
   # If the item is discounted, the current price is 
   # `price * (100 - discount rate) / 100`. Otherwise, it is same as the price.
   #
   # TODO: Implement instance method 'current_price'
-
+  def current_price()
+    if discounted?    
+      return @price * (100 - @discount_rate) / 100
```

Comment:
Since we have defined attribute accessor, you can use `price` instead of `@price`. Likewise for the other variables